content/blog/pulumi-components/index.md

```diff
@@ -0,0 +1,331 @@
+                bucket: this.bucket.id,
+                rules: [{
+                    applyServerSideEncryptionByDefault: {
+                        sseAlgorithm: "AES256",
+                    },
+                }],
+            }, { parent: this });
+        }
+
+        this.bucketName = this.bucket.id;
+
+        this.registerOutputs({
+            bucket: this.bucket,
+            bucketName: this.bucketName,
+        });
+    }
+}
+```
+
+### Sharing Components
+
+Pulumi's new Components capability is designed to make sharing infrastructure building blocks as easy as possible, whether you're publishing them to a Git repository or referencing them from a local folder in a monorepo.
+
+#### Publishing to Git
+
+Once you've authored your component, sharing it is as simple as pushing your code to a Git repository and optionally tagging a release. From there, anyone (including you) can pull it down and use it in their own Pulumi projects.
+
+For example, after pushing your S3BucketComponent to GitHub and tagging it as v1.0.0, others can consume it by running:
+
+```bash
+pulumi package add github.com/myorg/secure-s3-component@v1.0.0
+```
+
+Under the hood, Pulumi:
+
+1. Fetches your code from GitHub
+2. Auto-generates a local SDK from the component's schema
+3. Makes it available to your Pulumi program in your chosen language
+
+#### Referencing Components Locally
+
+For scenarios when you're not wanting to publish components publicly, you can reference local source code directly:
+
+```bash
+pulumi package add /path/to/local/secure-s3-component
+```
+
+Pulumi will identify the folder as a Pulumi component project, generate a local SDK, and make it available for import in your program—even if your consumer program is in a different language.
+
+### Consuming Components Across Languages
+
+The real power comes in how these components can now be consumed in any Pulumi language:
+
+{{< chooser language "typescript,python,csharp,go,java,yaml" >}}
+
+{{% choosable language typescript %}}
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import { S3BucketComponent } from "@myorg/secure-s3-component";
+
+const secureBucket = new S3BucketComponent("my-secure-bucket", {
+    bucketName: "my-company-secure-assets",
+    tags: {
+        Environment: "production",
+        Department: "engineering",
+    },
+});
+
+export const bucketName = secureBucket.bucketName;
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+import pulumi
+from myorg_secure_s3_component import S3BucketComponent
+
+secure_bucket = S3BucketComponent("my-secure-bucket", 
+    bucket_name="my-company-secure-assets",
+    tags={
+        "Environment": "production",
+        "Department": "engineering",
+    })
+
+pulumi.export("bucket_name", secure_bucket.bucket_name)
+```
+
+{{% /choosable %}}
+
+{{% choosable language csharp %}}
+
+```csharp
+using Pulumi;
+using MyOrg.SecureS3Component;
+
+class MyStack : Stack
```

Comment:
So we tend not to use the `Stack` inheritance API anymore and use the callback one, as in https://github.com/pulumi/templates/blob/master/csharp/Program.cs